src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -70,7 +72,12 @@ public UserService(UserDAO userDAO, UserPropertyDAO userPropertyDAO, UserRoleDAO
      * @param userId The User's ID
      * @return The updated User
      */
-    public User updateUserFieldsById(UserUpdateFields userUpdateFields, Integer userId) {
+    public User updateUserFieldsById(@NonNull User user, UserUpdateFields userUpdateFields, Integer userId) {
+        // must be updating oneself or be an admin
+        if (!user.hasUserRole(UserRoles.ADMIN) && !user.getUserId().equals(userId)) {
+            throw new NotAuthorizedException("Cannot update user.");
+        }
```

Comment:
We have two general practices that apply here. 
1. If you look at `Resource.validateAuthedRoleUser()`, you'll see what we usually do in this situation where a user is trying to do something they cannot.
2. We like to separate out auth logic at the Resource level since that's where we have permissions annotations. Here in the service class, we like to keep the logic limited to business rules.